### PR TITLE
Skill - EffectSpore

### PR DESCRIPTION
**Pull request by JesterWizard**
If attacked, this unit has a 50% chance of inflicting one of sleep, poison or silence on the foe.

### CODE REVIEW

File: Contants/Texts/Source/texts/Skills_BattleStatus.txt [hunk 1]
```diff
@@ -851,3 +851,11 @@ L-Brace[X]
 ## MSG_SKILL_LunarBrace
 Lunar Brace:[NL]
 Negate 25% of enemy defense.[X]
+
+## MSG_SKILL_EffectSpore_NAME
+E-Spore[X]
+
+## MSG_SKILL_EffectSpore
+Effect Spore:[NL]
+If attacked, 50% chance of inflicting[NL]
```
Comment (MokhaLeee) [inline Contants/Texts/Source/texts/Skills_BattleStatus.txt:860]: isn't this desc too long that may cause rtext overflow?

**Comment by MokhaLeee** [comment]
Your judgement is wrong, this should judge on gBattleTarget rather than defender. both actor or target can be the defneder.

File: Contants/Texts/Source/texts/Skills_BattleStatus.txt (hunk 1 above)
Comment (JesterWizard) [reply Contants/Texts/Source/texts/Skills_BattleStatus.txt:860]: image

**Comment by JesterWizard** [comment]
> Your judgement is wrong, this should judge on gBattleTarget rather than defender. both actor or target can be the defneder.

I disagree. If I do that it activates on both sides. As it is now, it only activates for the skill holder.

**Comment by MokhaLeee** [comment]
I think "if attacked" means as battle target?

**Comment by MokhaLeee** [comment]
If you want the battle-target also affected by battle-actor's skill effect, then you need to test what will happen when you try to poison a stoned unit.

**Comment by JesterWizard** [comment]
> I think "if attacked" means as battle target?

Yeah, but it triggers on both so I can't gBattleTarget. I can send you a video?

File: Wizardry/Core/BattleSys/Source/BattleHitAttr.c [hunk 2]
```diff
@@ -230,6 +230,47 @@ void BattleHit_InjectNegativeStatus(struct BattleUnit * attacker, struct BattleU
...
+        case UNIT_STATUS_CRIT:
+        case UNIT_STATUS_AVOID:
+            break;
+            
+        default:
+            return;
+        }
+        
+        RegisterTargetEfxSkill(GetBattleHitRound(gBattleHitIterator), SID_EffectSpore);
+        int effectID = NextRN_N(3);
```
Comment (MokhaLeee) [inline Wizardry/Core/BattleSys/Source/BattleHitAttr.c:256]: There should have no value declaration inside a block, mind this next time

**Pull request merged**